docs/en/operation/Server-Logging.md

```diff
@@ -0,0 +1,201 @@
+runtime.
+
+The Alluxio shell comes with a `logLevel` command that allows you to get or change the log level of a
+particular class on specific master or worker instances.
+
+The syntax is `alluxio logLevel --logName=NAME [--target=<master|worker|host:port>] [--level=LEVEL]`,
+where the `logName` indicates the logger's name, and `target` lists the Alluxio masters or
+workers to set. If parameter `level` is provided the command changes the logger level, otherwise it
+gets and displays the current logger level. `logName` typically represents the name of the class
+that is logging the information.
+
+For example, this command sets the class `alluxio.heartbeat.HeartbeatContext`'s logger level to
+DEBUG on master as well as a worker at `192.168.100.100:30000`.
+
+```bash
+alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=master,192.168.100.100:30000 --level=DEBUG
+```
+
+And the following command gets all workers' log level on class `alluxio.heartbeat.HeartbeatContext`
+```bash
+alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=workers
+```
+
+For more information, refer to the help text of the `logLevel` commmand by running `./bin/alluxio logLevel`
+
+## Remote Logging
+
+### Overview
+
+Alluxio supports sending logs to a remote log server over the network. This feature can be useful
+to system administrators who have to perform the task of log collection. With remote logging, the
+log files, e.g. master.log, worker.log, etc. on all Alluxio servers will be readily available on
+a designated and configurable directory on the log server.
+
+### Deploying The Log Server
+
+#### Configuring the Log Server
+
+You can choose the directory that the log server will write logs to by setting the
+`$ALLUXIO_LOGSERVER_LOGS_DIR` environment variable.
+
+#### Start Log Server
+
+On the log server, execute the following command.
+
+```bash
+$ ./bin/alluxio-start.sh logserver
+```
+
+#### Stop the Log Server
+
+```bash
+$ ./bin/alluxio-stop.sh logserver
+```
+
+### Configuring Alluxio Processes to use the Log Server
+
+By default, remote logging is not enabled. To enable Alluxio remote logging, you can set
+environment variables or modify Alluxio's `log4j.properties`
+
+There is no requirement on where the log server can be run, as long as the other Alluxio servers
+have network access to it. In our example, we run the log server on the same machine as a master.
+
+#### Enable Remote Logging with Environment Variables
+
+The three environment variables `ALLUXIO_LOGSERVER_HOSTNAME`, `ALLUXIO_LOGSERVER_PORT` and
+`ALLUXIO_LOGSERVER_LOGS_DIR` control Alluxio remote logging behavior.
+
+Suppose the hostname of the log server is `AlluxioLogServer`, and the port is `45010`.
+In `conf/alluxio-env.sh`, add the following lines:
+
+```bash
+ALLUXIO_LOGSERVER_HOSTNAME=AlluxioLogServer
+ALLUXIO_LOGSERVER_PORT=45010
+ALLUXIO_LOGSERVER_LOGS_DIR=/tmp/alluxio_remote_logs
```

Comment:
This is configured on the Alluxio master or worker, right? If so, does this line actually affect the log server logs dir?

Reply:
I don't think so. There are property keys `LOGSERVER_LOGS_DIR` and `LOGSERVER_HOSTNAME` but there aren't usages of them anywhere in the codebase as far as I can tell. And the way that the `alluxio-start.sh` script starts the logserver (`./bin/alluxio-start.sh logserver`) uses the `ALLUXIO_LOGSERVER_LOGS_DIR` environment variable which is propagated. You can follow the code path here.

Shell script
https://github.com/Alluxio/alluxio/blob/master/bin/alluxio-start.sh#L163-L176

LogServer
https://github.com/Alluxio/alluxio/blob/master/logserver/src/main/java/alluxio/logserver/AlluxioLogServer.java#L25-L29

LogServerProcess
https://github.com/Alluxio/alluxio/blob/master/logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java#L69-L76

`mBaseLogsDir` isn't set anywhere else apart from the method constructor and the constructor is only ever called from `AlluxioLogServer.java`

Reply:
I see. If this is the case, then we should remove this line from the recommended way to start alluxio processes, right?